### PR TITLE
08 refactor: the SearchPage design

### PR DESCRIPTION
**Pull request by paschalidi**
Hey @JoakimSM there is no ticket for this. I basically take a step back and doing a small refinement in the component. 

### What is this PR about?
In general I am renaming and kinda refactoring the component in my attempt to achieve a better and more readable design. 

On thing that I am doing is adding tests. Also you will find here the `.actions` file where all the `actionCreator` functions live.

Have a look and let me know what you think 🕊️

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js [hunk 1]
```diff
@@ -45,12 +45,14 @@ const Index = ({
     searchViaUniqueIdOnScopeProgram,
     searchViaAttributesOnScopeProgram,
     searchViaAttributesOnScopeTrackedEntityType,
+    saveCurrentFormData,
     selectedSearchScopeId,
     classes,
-    availableSearchOptions,
+    searchGroupForSelectedScope,
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js:51]: NIT: I would prefer this to be named searchGroup**s**ForSelectedScope, signaling there are multiple searchGroups here. This is nitpicking, I know, but this threw me off a bit.

File: src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.epics.js [hunk 2]
```diff
@@ -2,15 +2,15 @@
 import { catchError, flatMap, map, startWith } from 'rxjs/operators';
 import { from } from 'rxjs/observable/from';
 import { of } from 'rxjs/observable/of';
-import { searchPageActionTypes } from '../SearchPage.container';
+import { searchPageActionTypes } from '../SearchPage.actions';
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.epics.js:5]: perfect 👍

File: src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.container.js [hunk 3]
```diff
@@ -1,60 +1,87 @@
...
+    searchViaAttributesOnScopeProgram,
+    searchViaAttributesOnScopeTrackedEntityType,
+    searchViaUniqueIdOnScopeProgram,
+    searchViaUniqueIdOnScopeTrackedEntityType,
+} from '../SearchPage.actions';
+import { actionCreator } from '../../../../actions/actions.utils';
+
+const isValueContainingCharacter = string => string.replace(/\s/g, '').length;
+
+const collectCurrentSearchTerms = (searchGroupForSelectedScope, formsValues) => {
```
[the remark this answers is not in the document]
Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.container.js:16]: yeah that's a good idea ;) will do so

File: src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js [hunk 4]
```diff
@@ -8,12 +8,22 @@ import { withErrorMessageHandler, withLoadingIndicator } from '../../../HOC';
...
 import { addFormData } from '../../D2Form/actions/form.actions';
 import { actionCreator } from '../../../actions/actions.utils';
-
+import { searchPageActionTypes } from './SearchPage.actions';
 
 const buildSearchOption = (id, name, searchGroups, searchScope) => ({
     searchOptionId: id,
     searchOptionName: name,
     searchGroups: [...searchGroups.values()]
+        // We use the sorted array to always have expanded the first search group section.
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js:17]: @JoakimSM also look at this. Now it is being calculated here. This is from the https://github.com/dhis2/capture-app/pull/839#discussion_r462112144. I would prefer to change this in this PR and not in the previous one but you can let me know if this is a blocker for you and I can apply the changes in the previous PR :)

File: src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js (hunk 1 above)
Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js]: yeah thanks for this! :)

File: src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js (hunk 4 above)
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js:17]: Thanks. I was thinking we could move it up even one more level and sort the search groups when the metadata objects are built. Somewhere around here: https://github.com/dhis2/capture-app/blob/d779c06f20e0e5a6ab23b938d9ffe294c7f405e4/src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js#L197

Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js:17]: Dont see why not :) Will do so then in the https://github.com/dhis2/capture-app/pull/839#discussion_r462112144.

Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js:17]: I would appreciate some of your help here. :) 
How would you go about it, would you do something like the image bellow? Or am I missing a point?
image

**Comment by paschalidi** [comment]
@JoakimSM have looked into the comment and applied the changes you suggested! Thanks again :)

File: src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js (hunk 4 above)
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js]: LGTM 👍

**Pull request merged**